Remove doctest dependency on ethtoken

stefanmendoza opened the pull request:
### What was wrong?

ethtoken has a dependency on web3 and web3 depends on ethtoken for doctest. We should remove this circular dependency and just include the EIP20 ABI in the doctest for now.

See: https://github.com/carver/ethtoken.py/pull/4#discussion_r246949117

### How was it fixed?

Remove ethtoken dependency and added the EIP20 ABI to the relevant test.

#### Cute Animal Picture

![Put a link to a cute animal picture inside the parenthesis-->](https://www.wowamazing.com/wp-content/uploads/2015/06/cute-miniature-horse-4.jpg)

pipermerriam commented on docs/web3.eth.account.rst:
```diff
@@ -259,9 +259,11 @@ To sign a transaction locally that will invoke a smart contract:
+    >>> import json
     >>> from web3.auto import w3
 
+    >>> EIP20_ABI = json.loads('[{"constant":true,"inputs":[],"name":"name","outputs":[{"name":"","type":"string"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":false,"inputs":[{"name":"_spender","type":"address"},{"name":"_value","type":"uint256"}],"name":"approve","outputs":[{"name":"","type":"bool"}],"payable":false,"stateMutability":"nonpayable","type":"function"},{ ...
```
It'd be nice to move this into a `.. testsetup::` directive so that the `EIP20_ABI` variable is available in the `doctest` code blocks but such that we don't have to have this very large JSON string in our code block.  See the examples RST file for an example of how we do this elsewhere.

stefanmendoza replied:
@carver getting back to this...

Since ethtoken was removed, did you literally mean link to the Github URL in a comment?

kclowes commented:
@stefanmendoza correct me if I'm wrong, but it looks like this is waiting for the ABI to be converted into a `.. testsetup::` directive? Do you think you'll have time to get to that? I'm happy to do it if not, just let me know!

stefanmendoza commented:
@kclowes feel free to push this over the finish line! Sorry, I got pulled into other things and this got put on the back burner. Thanks!

Also, thanks to @carver and @pipermerriam. I appreciate the review.

@kclowes  - Lemme know if you have any questions for me! 🙏

kclowes commented:
Sounds good! No worries!

stefanmendoza commented:
I had deleted my fork, so I recovered the commit from this PR and addressed the last change request. The new PR can be found here:
https://github.com/ethereum/web3.py/pull/1395
